app/root.tsx

```diff
@@ -46,6 +47,11 @@ import { useNonce } from './utils/nonce-provider.ts'
 import { makeTimings, time } from './utils/timing.server.ts'
 import { useToast } from './utils/useToast.tsx'
 import { useOptionalUser, useUser } from './utils/user.ts'
+import rdtStylesheetUrl from 'remix-development-tools/stylesheet.css'
+const RemixDevTools =
+	process.env.NODE_ENV === 'development'
+		? lazy(() => import('remix-development-tools'))
+		: undefined
```

Comment:
There is a way to do this already. There is a requireUrlFlag on the component which when you pass only renders the dev tools when ?rdt=true is present in the URL, but this would require you to increase the bundle size of your application. It can already be done but there is a trade-off. Up to you to decide, it is your stack after all!

Reply:
I added it to v1.1.1 which this is so you can try it out and see if you like it! @kentcdodds I might've implemented it after your tweet pre-emptively

Reply:
@kentcdodds I will take a deep dive when I get the time into this, this might be the way to go in the future releases

Reply:
👍👍 Sounds good. Thanks for working on this!

Reply:
@kentcdodds I really like your approach here, any chances you have an article on bundling and what would get omitted and what wouldn't, with some caveats and foot-guns. I would love to do it this way I am just afraid of something ending up in the users production bundle

Reply:
also my issue with this approach is that what you're doing is client only, but remix is server + client, so a bit harder to just wrap it like this

Reply:
That's fine. Let's get this in and we can iterate to something better later.